### ISSUE TITLE
Validation is not working in the "Message All Volunteers" modal

### ISSUE THREAD

**Issue by nemesv** [HTBox/allReady#1411]
When you try to create an invalid message, without a subject and message, "nothing happens": 
- no error message
- the modal stays open
- the "Send Message" gets disabled

The following JS error is displayed in the console: 

```
VM209:1 Uncaught SyntaxError: Unexpected token B in JSON at position 0(…)
error @ taskDetails.js:40
```

![image](https://cloud.githubusercontent.com/assets/251330/19819945/40d214ce-9d57-11e6-90da-b7da1571fbf9.png)

Repro Steps.
1. Log into Org admin That has an Event created.
2. Add volunteers to event task.
3. Navigate to Task Details /admin/task/details.
4. Click Message All Button.
5. Leave all inputs empty
6. Click Send Message Button

**Comment by adam3039** [comment]
Doing this!

**Comment by dcbrewster** [comment]
adam3039. Are you still working on this?

**Comment by tonysurma** [comment]
@dcbrewster it's been long enough that I would consider this not work in progress by @adam3039 and feel free to start working on it.  Thanks.

**Comment by stevejgordon** [comment]
@dcbrewster - Did you start working on this in the end?

**Comment by dcbrewster** [comment]
Yes. It is done. Just need the other pull request (#1676) to be implemented before I commit these changes\

Dave

**Comment by stevejgordon** [comment]
@dcbrewster - Great! Thanks for the swift reply too.

**Comment by naeemsarfraz** [comment]
I've just submitted PR #1723 which adds a required attribute to the input elements which should stop an empty form being posted.

Looking into providing feedback to the user about what's gone wrong.

**Comment by BillWagner** [comment]
Fixed in PR #1723
